core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java

```diff
@@ -251,38 +251,69 @@ public void importData() throws IOException, DaoException {
-					    	continue;
-					    }				    	
-					}
-				}
+                //  Assume we are dealing with Entrez Gene Ids (this is the best / most stable option)
+                String geneSymbol = record.getHugoGeneSymbol();
+                String entrezIdString = record.getGivenEntrezGeneId();
+
+                CanonicalGene gene = null;
+                if (!(entrezIdString.isEmpty() ||
+                      entrezIdString.equals("0"))) {
+                    Long entrezGeneId;
+                    try {
+                        entrezGeneId = Long.parseLong(entrezIdString);
+                    } catch (NumberFormatException e) {
+                        entrezGeneId = null;
+                    }
+                    if (entrezGeneId == null) {
+                        ProgressMonitor.logWarning(
+                                "Ignoring line with invalid Entrez_Id " +
+                                entrezIdString);
+                        entriesSkipped++;
+                        continue;
+                    } else {
+                        gene = daoGene.getGene(entrezGeneId);
+                        if (gene == null) {
+                            //skip
+                            ProgressMonitor.logWarning(
+                                    "Entrez gene ID " + entrezGeneId +
+                                    " not found. Record will be skipped.");
+                            entriesSkipped++;
+                            continue;
+                        }
+                    }
+                }
 
-				if(gene == null) {
-					// If Entrez Gene ID Fails, try Symbol.
-					gene = daoGene.getNonAmbiguousGene(geneSymbol, chr);
-				}
-                                
-				if(gene == null) {
-					ProgressMonitor.logWarning("Gene not found:  " + geneSymbol + " ["+ record.getGivenEntrezGeneId() + "] or ambiguous alias. Ignoring it "
-					                    + "and all mutation data associated with it!");
-					entriesSkipped++;
-					continue;
+                // If Entrez Gene ID Fails, try Symbol.
+                if (gene == null &&
+                        !(geneSymbol.equals("") ||
+                          geneSymbol.equals("Unknown"))) {
+                    gene = daoGene.getNonAmbiguousGene(geneSymbol, chr);
+                }
+
+                // assume Unknown / 0 to imply an intergenic irrespective of
+                // what the column Variant_Classification says
+                if (geneSymbol.equals("Unknown") &&
+                        entrezIdString.equals("0") &&
+                        mutationType != null &&
+                        !mutationType.equalsIgnoreCase("IGR")) {
+                    ProgressMonitor.logWarning(
+                            "Treating mutation with gene symbol 'Unknown' " +
+                            "and Entrez gene ID 0 as intergenic ('IGR') " +
+                            "instead of '" + mutationType + "'.");
+                    mutationType = "IGR";
```

Comment:
let's do something like 
```
myMutationFilter.igrRejects++;
entriesSkipped++;
continue;
```
and keep `if` below simple (just `if (gene == null)` )